test/functional/test_runner.py

```diff
@@ -728,42 +729,50 @@ def get_next(self):
             test_argv = test.split()
             testdir = "{}/{}_{}".format(self.tmpdir, re.sub(".py$", "", test_argv[0]), portseed)
             tmpdir_arg = ["--tmpdir={}".format(testdir)]
-            self.jobs.append((test,
+
+            def proc_wait(task):
+                task[2] = task[2].wait()
```

Comment:
I find this confusing - as mentioned below -, we're replacing a value with a different typed one. Is the mutation strictly necessary or would something like
```python
def proc_wait(task):
    name, start_time, proc, testdir, log_out, log_err = task
    return_code = proc.wait()
    return [name, start_time, return_code, testdir, log_out, log_err]
```

also work?